src/Microsoft.Health.Fhir.Api.UnitTests/Controllers/AadSmartOnFhirProxyControllerTests.cs

```diff
@@ -0,0 +1,118 @@
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Routing;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Api.UnitTests.Controllers
+{
+    public class AadSmartOnFhirProxyControllerTests
+    {
+        private readonly SecurityConfiguration _securityConfiguration = new SecurityConfiguration();
+        private readonly IHttpClientFactory _httpClientFactory = Substitute.For<IHttpClientFactory>();
+        private readonly ILogger<Core.Configs.SecurityConfiguration> _logger = NullLogger<Core.Configs.SecurityConfiguration>.Instance;
+        private readonly IUrlResolver _urlResolver = Substitute.For<IUrlResolver>();
+        private readonly FeatureConfiguration _featureConfiguration = new FeatureConfiguration();
+        private readonly AadSmartOnFhirProxyController _controller;
+
+        public AadSmartOnFhirProxyControllerTests()
+        {
+            _securityConfiguration.EnableAadSmartOnFhirProxy = true;
+            _securityConfiguration.Enabled = true;
+            _securityConfiguration.Authorization = new AuthorizationConfiguration
+            {
+                Enabled = true,
+            };
+            _securityConfiguration.Authentication = new AuthenticationConfiguration
+            {
+                Audience = "testaudience",
+                Authority = "http://testauthority.com/v2.0",
+            };
+
+            string openIdConfiguration;
+            using (StreamReader r = new StreamReader(Assembly.GetExecutingAssembly().
+                GetManifestResourceStream("Microsoft.Health.Fhir.Api.UnitTests.Controllers.openid-configuration.json")))
+            {
+                    openIdConfiguration = r.ReadToEnd();
+            }
+
+            var httpMessageHandler = new TestHttpMessageHandler(new HttpResponseMessage()
+            {
+                StatusCode = System.Net.HttpStatusCode.OK,
+                Content = new StringContent(openIdConfiguration),
+            });
+
+            var httpClient = new HttpClient(httpMessageHandler);
+
+            _httpClientFactory.CreateClient().Returns(httpClient);
+
+            _controller = new AadSmartOnFhirProxyController(
+                Options.Create(_securityConfiguration),
+                _httpClientFactory,
+                _urlResolver,
+                _logger);
+        }
+
+        [Theory]
+        [InlineData(null, null, null, null, null, null, null)]
+        [InlineData("code", null, null, null, null, null, null)]
+        [InlineData("code", "clientId", null, null, null, null, null)]
+        [InlineData("code", "clientId", "https://testurl.com", null, null, null, null)]
+        [InlineData("code", "clientId", "https://testurl.com", "launch", null, null, null)]
+        [InlineData("code", "clientId", "https://testurl.com", "launch", "scope", null, null)]
+        [InlineData("code", "clientId", "https://testurl.com", "launch", "scope", "state", null)]
```

Comment:
Do we need a case to test when the `!_isAadV2 && !string.IsNullOrEmpty(aud)` if block is executed?